packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/utils.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+
+/// Encode values of [metadata].
+///
+/// A value that contains non-ascii chars will be encoded.
+/// Details see [Working with object metadata](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html).
+Map<String, String> encodeMetadata(Map<String, String> metadata) {
+  return Map.fromEntries(
+    metadata.entries.map((e) => MapEntry(e.key, _rfc2047UTF8Encode(e.value))),
+  );
+}
+
+/// Decode values of [metadata].
+///
+/// A value that is a utf8 encoded string will be decoded.
+Map<String, String> decodeMetadata(Map<String, String> metadata) {
+  return Map.fromEntries(
+    metadata.entries.map((e) => MapEntry(e.key, _rfc2047UTF8Decode(e.value))),
+  );
+}
+
+bool _shouldEncode(String str) {
```

Comment:
```suggestion
bool _isUtf8(String str) {
```

Reply:
Agreed, changed the function name to `_hasNonAscii`.